src/CSV.jl

```diff
@@ -38,6 +38,9 @@ end
     return byte
 end
 
+substitute(::Type{Union{T, Missing}}, ::Type{T1}) where {T, T1} = Union{T1, Missing}
```

Comment:
Actually, that's the kind of function that I wanted to have in `Missings`/0.7 Base, because it's implicitly used in several places at JuliaData (cc @nalimilan).
Maybe it should be called `substitute_type`?

Reply:
In Base the convention seems to be `type*`, e.g. `typeintersect` and `typejoin` (we have discussed adding `typesubtract`). So why not `typesubstitute` (but of course it's much less general than the others).

Reply:
It's true, for the Base it should be more specific. `substitute_nonmissing_type`? `nonmissingtype_substitute`? Don't have a good variant that starts with `type`

Reply:
I'm not sure we'll be able to include all of these helper functions in Base, so for now it could be just `Missings.typesubstitute` or something like that.